pennylane/template.py

```diff
@@ -0,0 +1,225 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=protected-access
+r"""
+QML Templates
+=============
+
+**Module name:** :mod:`pennylane.template`
+
+.. currentmodule:: pennylane.template
+
+This module provides a growing library of templates of common quantum
+machine learning circuit architectures that can be used to easily build
+more complex quantum machine learning models.
+
+For example, you can construct, evaluate, and train quantum circuit based
+on the circuit-centric quantum classifier architecture from
+:cite:`schuld2018circuit` on an arbitrary number of wires and with an
+arbitrary number of blocks in the following way:
+
+.. code-block:: python
+
+    import pennylane as qml
+    from pennylane import numpy as np
+
+    num_wires=4
+    num_blocks=3
+    dev = qml.device('default.qubit', wires=num_wires)
+
+    @qml.qnode(dev)
+    def circuit(weights, x=None):
+        qml.BasisState(x, wires=range(num_wires))
+        qml.template.CircuitCentric(weights, periodic=True, wires=range(num_wires))
+        return qml.expval.PauliZ(0)
+
+    weights=np.random.randn(num_blocks, num_wires, 3)
+    print(circuit(weights, x=np.array(np.random.randint(0, 1, num_wires))))
+
+
+The handy :func:`Interferometer` function can be used to construct arbitrary
+interferometers in terms of elementary :class:`~.Beamsplitter` Operations.
+operations, by giving a lists of beamsplitter parameters. PennyLane can then be used to
```

Comment:
typo: "a lists" --> "lists" or "a list" (not sure which one is intended)

Reply:
replaced by "two lists"